tests/python/shared/fixtures/init.py

```diff
@@ -185,12 +185,30 @@ def delete_compose_files():
         filename.unlink(missing_ok=True)
 
 
+def wait_for_services():
+    wait_for_server()
+
+
 def wait_for_server():
-    for _ in range(30):
-        response = requests.get(get_api_url("users/self"))
-        if response.status_code == HTTPStatus.UNAUTHORIZED:
-            break
-        sleep(5)
+    for i in range(300):
+        logging.getLogger(__package__).debug(f"waiting for the server to appear ... ({i})")
+        response = requests.get(get_server_url("api/server/health/", format="json"))
+        try:
+            statuses = response.json()
+            if all(v == "working" for v in statuses.values()):
```

Comment:
@zhiltsov-max , I believe it is better just to check the response status. If it is 200, the server is fine.

Reply:
Changed